### PR TITLE
PIL LogUp implementation: Generate multiplicity column

### PR DESCRIPTION
**Pull request by georgwiese**
No need to have the user create the multiplicity column anymore.

### CODE REVIEW

File: executor/src/witgen/machines/fixed_lookup_machine.rs [hunk 1]
```diff
@@ -174,7 +174,7 @@ impl<T: FieldElement> IndexedColumns<T> {
     }
 }
 
-const MULTIPLICITY_LOOKUP_COLUMN: &str = "m_logup_multiplicity";
+const MULTIPLICITY_LOOKUP_COLUMN: &str = "multiplicities";
```
[the remark this answers is not in the document]
Comment (georgwiese) [reply executor/src/witgen/machines/fixed_lookup_machine.rs:177]: #2043

File: std/protocols/lookup.asm [hunk 2]
```diff
@@ -62,9 +62,8 @@ let compute_next_z: Fp2<expr>, Fp2<expr>, Fp2<expr>, Constr, expr -> fe[] = quer
 /// the used challenges would overlap.
 /// TODO: Implement this for an array of constraints.
 /// Arguments:
-/// - lookup_constraint: The lookup constraint
-/// - multiplicities: A multiplicities column which shows how many times each row of the RHS value appears in the LHS                  
-let lookup: Constr, expr -> () = constr |lookup_constraint, multiplicities| {
+/// - lookup_constraint: The lookup constraint
```
Comment (chriseth) [inline std/protocols/lookup.asm:65]: Quite redundant, I would just remove it.

File: std/protocols/lookup.asm [hunk 3]
```diff
@@ -75,6 +74,7 @@ let lookup: Constr, expr -> () = constr |lookup_constraint, multiplicities| {
 
     let lhs_denom = sub_ext(beta, fingerprint(lhs, alpha));
     let rhs_denom = sub_ext(beta, fingerprint(rhs, alpha));
+    let multiplicities = std::prover::new_witness_col_at_stage("multiplicities", 0);
```
Comment (chriseth) [inline std/protocols/lookup.asm:77]: Why specify the stage explicitly?

Comment (chriseth) [reply std/protocols/lookup.asm:77]: If you use `let multiplicities`, it will be create at the "current stage".

Comment (chriseth) [reply std/protocols/lookup.asm:77]: Right, if we use `capture_stage`, then we might already be at a higher stage, but I think there should still be a better way around that than just setting the stage to zero explicitly. I think I would just use `let multiplicites` now until we reach that problem.

File: executor/src/witgen/mod.rs [hunk 4]
```diff
@@ -281,7 +281,9 @@ impl<'a, 'b, T: FieldElement> WitnessGenerator<'a, 'b, T> {
             .filter(|(symbol, _)| symbol.stage.unwrap_or_default() <= self.stage.into())
             .flat_map(|(p, _)| p.array_elements())
             .map(|(name, _id)| {
-                let column = columns.remove(&name).unwrap();
+                let column = columns
+                    .remove(&name)
+                    .unwrap_or_else(|| panic!("No machine generated witness for column: {name}"));
```
[the remark this answers is not in the document]
Comment (georgwiese) [reply executor/src/witgen/mod.rs:286]: I think the culprit is [this line](https://github.com/powdr-labs/powdr/blob/a262b36d40042e9a6fc113036ab22e9dac30fba1/executor/src/witgen/machines/machine_extractor.rs#L197), no one feels responsible for it. Hoping to fix this soon in #2043.

File: std/protocols/lookup.asm (hunk 2 above)
Comment (georgwiese) [reply std/protocols/lookup.asm]: done

File: std/protocols/lookup.asm (hunk 3 above)
Comment (georgwiese) [reply std/protocols/lookup.asm]: done

**Review by leonardoalt — APPROVED** [review]

**Review by chriseth — APPROVED** [review]
I still don't understand the panic message but it's fine.

**Pull request merged**